[DOCS] Add 'Using ES|QL in Elastic Security' (#101677)
elastic/elasticsearch: docs/reference/esql/esql-security-solution.asciidoc

[[esql-elastic-security]]
=== Using {esql} in {elastic-sec}

++++
<titleabbrev>Using {esql} in {elastic-sec}</titleabbrev>
++++

You can use {esql} in {elastic-sec} to investigate events in Timeline and create
detection rules. Use the Elastic AI Assistant to build {esql} queries, or answer
questions about the {esql} query language.

[discrete]
[[esql-elastic-security-timeline]]
=== Use {esql} to investigate events in Timeline

You can use {esql} in Timeline to filter, transform, and analyze event data
stored in {es}. To start using {esql}, open the the **{esql}** tab. To learn
more, refer to {security-guide}/timelines-ui.html[Investigate events in
Timeline].

[discrete]
[[esql-elastic-security-detection-rules]]
=== Use {esql} to create detection rules

Use the {esql} rule type to create detection rules using {esql} queries. The
{esql} rule type supports aggregating and non-aggregating queries. To learn

> more, refer to {security-guide}/rules-ui-create.html[Create a detection rule].

Reviewer: ```suggestion
more, refer to {security-guide}/rules-ui-create.html#create-esql-rule[Create an {esql} rule].
```
